<?xml version="1.0"?>
<launch>
  <arg name="image_selector" value="True"/>
  <arg name="image_broadcaster" value="False"/>
  <arg name="meter_per_pixel" value="0.003"/>

  <!-- select a single vector field -->
  <group if="$(arg image_selector)">
    <node name="image_selecter_node" pkg="vector_field_navigation" type="image_selecter_node" output="screen" cwd="ROS_HOME">
      <param name="image_publisher_topic" value="/image"/>
    </node>
  </group>

  <!-- change periodically the vector field and switch through images in folder -->
  <group if="$(arg image_broadcaster)">
    <node pkg="vector_field_navigation" name="repeat_images" type="ImageBroadcaster.py" output="screen">
      <param name="topic" value="/image"/>
      <param name="delay" value="30"/>
      <param name="path" value="$(find potential_field_navigation)/vectorfields"/>
    </node>
  </group>

  <node name="image_to_vectorfield_node" pkg="vector_field_navigation" type="image_to_vectorfield_node" output="screen" cwd="ROS_HOME">
    <param name="image_listener_topic" value="/image"/>
    <param name="potentialfield_publisher_topic" value="/potentialfield/image"/>
    <param name="vectorfield_publisher_topic" value="/vectorfield/image"/>
    <param name="heuristic_abs_min" value="0.05"/> <?Undefined An arbitrary value which indecates the start of value decay?>
    <param name="heuristic_factor" value="1.0"/> <?Maximum vector lengh?>
    <param name="heuristic_apply" value="1"/> <?Use heuristic instead of native vectorfield?>
    <param name="speed_scale" value="0.01"/>
  </node>

  <node name="vectorfield_to_gridmap_node" pkg="vector_field_navigation" type="vectorfield_to_gridmap_node" output="screen" cwd="ROS_HOME">
    <param name="meter_per_pixel" value="$(arg meter_per_pixel)"/>
    <param name="frame_id" value="world"/>
    <param name="vectorfield_listener_topic" value="/vectorfield/image"/>
    <param name="gridmap_publisher_topic" value="/vectorfield/image/gridmap"/>
  </node>
</launch>